semgrep/semgrep/pattern_match.py

```diff
@@ -37,7 +37,7 @@ def extra(self) -> Dict[str, Any]:
     @property
     def vars(self) -> Dict[str, Any]:
         metavars = {v: data.get("unique_id", {}) for v, data in self.metavars.items()}
-        return {v: uid.get("sid", "md5sum") for v, uid in metavars.items()}
+        return {v: uid.get("sid", uid.get("md5sum")) for v, uid in metavars.items()}
```

Comment:
Not sure if this was actually causing a problem, but it looked like a bug to me. Variables returned from `semgrep-core` either have a `sid` or `md5sum`. In this case, if the variable didn't have a `sid` it was using the literal `"md5sum"` instead of the value of that key. This seems like it should've broken metavariable matching, but I didn't look into it too much.

Reply:
Oh yeah, I'm not sure what the best failure case here is if the result has neither the `sid` nor the `md5sum`. Is that possible? Do we bail on the whole run, or just skip that variable?

Reply:
Interesting, the test failures are actually because this surfaced a bug in one of our rules that was unintentionally making use of metavariable matching.

Reply:
Probably ok to raise an exception. idk if we want to show work that was done or just bail completely?

Reply:
Per the two `semgrep-rules` that needed fixing, I'd say this definitely was a bug. I added some code to test if anything hits that case. Let's see what happens :+1: